lib/gc/ExecutionEngine/CPURuntime/ThreadLocals.cpp

```diff
@@ -0,0 +1,114 @@
+//===-- ThreadLocals.cpp - threadlocal ---------------------------------*- C++
+//-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "gc/ExecutionEngine/MemoryPool/ThreadLocals.h"
+#include <functional>
+#include <list>
+#include <mutex>
+#include <utility>
+#include <vector>
+
+namespace mlir {
+namespace gc {
+
+// if registry is destoryed, it will be set to true
+static bool registry_destroyed = false;
+
+thread_local_registry_t::thread_local_registry_t() = default;
+
+// release all registered TLS resources. The registry still keeps track of
+// the TLS objects
+void thread_local_registry_t::release() {
```

Comment:
currently we don't need this. I suggest to remove it and add in Memorypool.cpp

```
static thread_local  memory_pool::filo_memory_pool_t main_memory_pool_{
      memory_pool::main_chunk_size};
  // if the current thread is a worker thread, use this pool
static thread_local  memory_pool::filo_memory_pool_t thread_memory_pool_{
      memory_pool::threadlocal_chunk_size};
```

Reply:
removed `thread_local_buffer_t`